src/prometheus-net.DotNetRuntime/StatsCollectors/ExceptionStatsCollector.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Prometheus.DotNetRuntime.EventSources;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Text;
+#if PROMV2
+using Prometheus.Advanced;
+#endif
+
+
+namespace Prometheus.DotNetRuntime.StatsCollectors
+{
+    public class ExceptionStatsCollector : IEventSourceStatsCollector
+    {
+        private const int EventIdExceptionThrown = 80;
+        private const string LabelReason = "exception";
+
+        internal Counter NumExceptions { get; private set; }
+        internal Counter ExceptionReasons { get; private set; }
+
+        public Guid EventSourceGuid => DotNetRuntimeEventSource.Id;
+
+        public EventKeywords Keywords => (EventKeywords)DotNetRuntimeEventSource.Keywords.Exception;
+        public EventLevel Level => EventLevel.Informational;
+
+        public void RegisterMetrics(MetricFactory metrics)
+        {
+            NumExceptions = metrics.CreateCounter(
+                "dotnet_num_exceptions",
+                "Total of exceptions"
+            );
+
+            ExceptionReasons = metrics.CreateCounter(
+                "dotnet_exception_reasons",
+                "Reasons that led to an exception",
+                LabelReason
+            );
+        }
+
+        public void UpdateMetrics()
+        {
+
+        }
+
+        public void ProcessEvent(EventWrittenEventArgs e)
+        {
+            if (e.EventId == EventIdExceptionThrown)
+            {
+                NumExceptions.Inc();
+                ExceptionReasons.Labels((string)e.Payload[1]).Inc();
```

Comment:
What does the reason payload contain? If it's the full exception string representation (e.g. `Type` + `Message` + `StackTrace`), then this isn't something that appropriate for a prometheus label.

Reply:
Hi @djluck .
This payload are dvide in two parts:
[1] = namespace the exception: System.DivideByZeroException
[2] = exception message: Attempted to divide by zero.

In the label, I only inform the exception message. 
If it is still a bad practice, please indicate the best one

Reply:
I'm fine with just including the full name of the type, I was just making sure we weren't including the exception message- as some exception types include have a great deal of variation in their message (e.g. connection ids, invalid values, timeout durations, etc) they would explode label cardinality.